libs/header/p2p/session.go

```diff
@@ -174,6 +176,7 @@ func (s *session[H]) doRequest(
 
 		select {
 		case <-s.ctx.Done():
+			return
```

Comment:
Nice catch

Reply:
Wait didn't I mention this in one of the PRs?

Reply:
It seems that it was missed during resolving merge conflicts 😞